### PR TITLE
ensure no subpixel transforms on marker elements

### PR DESCRIPTION
**Pull request by mollymerp**
## Launch Checklist

While subpixel transforms turned out not to be the issue on popups (#3249 ), they are an issue with markers. Here I just round the `Point` pos for the marker (like we do with [`Popup`](https://github.com/mapbox/mapbox-gl-js/blob/master/js/ui/popup.js#L231) to prevent the blurriness it causes. 

_subpixel transform_
screen shot 2016-09-28 at 5 16 27 pm

_rounded pixel transform_
screen shot 2016-09-28 at 5 17 03 pm

cc @jfirebaugh @lucaswoj 

 - [x] briefly describe the changes in this PR
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page

### CODE REVIEW

**Comment by mourner** [comment]
As far as I remember, I didn't add rounding because markers noticeably wiggled when zooming. Is this the case now? Maybe we should only round the position on moveend.

**Comment by mollymerp** [comment]
@mourner -- thanks for flagging. I didn't remember that detail about the shaky zooming. I modified so it only rounds the transform on moveend.

File: js/ui/marker.js [hunk 1]
```diff
@@ -148,9 +149,9 @@ Marker.prototype = {
         else popup.addTo(this._map);
     },
 
-    _update: function () {
+    _update: function (e) {
         if (!this._map) return;
-        var pos = this._map.project(this._lngLat)._add(this._offset);
+        var pos =  (e && e.type === "move") ? this._map.project(this._lngLat)._add(this._offset) : this._map.project(this._lngLat)._add(this._offset).round();
```
Comment (lucaswoj) [inline js/ui/marker.js]: This line could be a little more DRY

```
var pos = this._map.project(this._lngLat)._add(this._offset);
if (options.roundPos) pos = pos.round();
```

File: js/ui/marker.js [hunk 2]
```diff
@@ -44,6 +44,7 @@ Marker.prototype = {
         this._map = map;
         map.getCanvasContainer().appendChild(this._element);
         map.on('move', this._update);
+        map.on('moveend', this._update);
         this._update();
```
Comment (lucaswoj) [inline js/ui/marker.js]: We should round the coords upon marker initialization too. 

Perhaps we could do this by adding a `roundPos` option to `Marker#_update`
```js
map.on('move', this._update.bind(this));
map.on('moveend', this._update.bind(this, {roundPos: true}));
this._update({roundPos: true});
```

Comment (mollymerp) [inline js/ui/marker.js]: As the PR stands, it does round coords on initialization (only keeps un-rounded coords on `move`). We can add this option in the interest of more explicit/understandable code 👍

**Comment by lucaswoj** [comment]
Looks good! 🚢 on 🍏

**Pull request merged**